core/store/src/db/refcount.rs

```diff
@@ -84,22 +85,20 @@ pub fn decode_value_with_rc(bytes: &[u8]) -> (Option<&[u8]>, i64) {
     }
 }
 
-/// Encodes value with reference count.
+/// Encode a positive reference count into the value.
+pub(crate) fn add_positive_refcount(data: &[u8], rc: std::num::NonZeroU32) -> Vec<u8> {
+    let rc = std::num::NonZeroI64::from(rc).get();
+    let mut value = Vec::with_capacity(data.len() + 8);
+    value.extend_from_slice(data);
+    value.extend_from_slice(&rc.to_le_bytes());
+    value
+}
+
+/// Returns empty value with encoded negative reference count.
 ///
-/// Returned value depends on the reference count.
-/// - rc = 0 ⇒ an empty vector,
-/// - rc < 0 ⇒ reference count encoded in little endian,
-/// - rc > 0 ⇒ data followed by the reference count encoded in little endian.
-pub(crate) fn encode_value_with_rc(data: &[u8], rc: i64) -> Vec<u8> {
-    if rc == 0 {
```

Comment:
Nice how `rc==0` is now statically unreachable! 

I wonder if `decode_value_with_rc` should also be changed to return NonZero?

Reply:
We would need to panic in decode_value_with_rc then.  In practice we shouldn’t be ever called with empty value or value with rc == 0 but using NonZero wouldn’t simplify any other code so it’s not really worth it.